### PR TITLE
Update remix package to use new GraphQL client format

### PR DESCRIPTION
**Pull request by paulomarg**
### WHY are these changes introduced?

As part of https://github.com/Shopify/shopify-api-js/pull/1068, `shopify-api` will now be exposing a new API for the GraphQL clients, that enables us to automatically type requests' variables and responses using `@shopify/api-codegen-preset`.

### WHAT is this pull request doing?

Preparing the Remix package to use the new client format exposed by the library, and ensure its outgoing types match the clients' so that the automatic typing still works the same way.

This only affects the outgoing types but it defaults them to the previous format, so this is not a breaking change.

## Type of change

- [x] Minor: New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have used `yarn changeset` to create a draft changelog entry (do NOT update the `CHANGELOG.md` files manually)
- [x] I have added/updated tests for this change
- [x] I have documented new APIs/updated the documentation for modified APIs (for public APIs)

### CODE REVIEW

**Review by byrichardpowell — APPROVED** [review]
Just a few questions.  Approving so you don't need a second review.

File: packages/shopify-app-remix/src/server/clients/storefront/factory.ts
Comment (paulomarg) [reply packages/shopify-app-remix/src/server/clients/storefront/factory.ts:24]: Worked like a charm!

File: .changeset/thirty-apples-think.md
Comment (paulomarg) [reply .changeset/thirty-apples-think.md]: Added a page with these details!

File: packages/shopify-app-remix/docs/staticPages/examples/guides/graphql-types/.graphqlrc.ts
Comment (lizkenyon) [inline packages/shopify-app-remix/docs/staticPages/examples/guides/graphql-types/.graphqlrc.ts:11]: '2024-01' ?

**Review by lizkenyon — APPROVED** [review]

**Pull request merged**
